src/app/components/heimdall-roles/heimdall-roles.component.ts

```diff
@@ -26,25 +25,24 @@ export class HeimdallRolesComponent implements OnInit {
   });
 
   constructor(private service: RoleService,
-              private snackBar: SnackBarService,
               private dialog: MatDialog,
               private roleService: RoleService,
               private realmService: RealmService) {
   }
 
   ngOnInit(): void {
-    this.subSink.add(this.realmService.realm.subscribe((data: ParentRealm) => {
-      this.allRoles = data.roles;
-      this.realm = data.realm;
+    this.subSink.add(this.realmService.realm$.subscribe((realm: Realm) => {
+      this.realm = realm;
+      this.getAllRoles();
     }));
   }
 
-  ngOnDestroy() {
-    this.subSink.unsubscribe();
+  getAllRoles() {
+    return this.subSink.add(this.roleService.getAllRoles(this.realm.name).subscribe(roles => this.allRoles = roles));
```

Comment:
formateaza corect

Reply:
ai deja un subsink.add in ngOnInit, aici nu ar trebui doar sa se returneze apelul la service?